core/src/test/java/org/owasp/dependencycheck/utils/InterpolationUtilTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of dependency-check-core.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright (c) 2023 Jeremy Long. All Rights Reserved.
+ */
+package org.owasp.dependencycheck.utils;
+
+import java.util.Properties;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Jeremy Long
+ */
+public class InterpolationUtilTest {
+    
+
+
+    /**
+     * Test of interpolate method, of class InterpolationUtil.
+     */
+    @Test
+    public void testInterpolate() {
+        Properties prop = new Properties();
+        prop.setProperty("key", "value");
+        prop.setProperty("nested", "nested ${key}");
+        String text = "This is a test of '${key}' '${nested}'";
+        String expResults = "This is a test of 'value' 'nested value'";
+        String results = InterpolationUtil.interpolate(text, prop);
+        assertEquals(expResults, results);
+    }
+    
```

Comment:
```suggestion

    @Test
    public void testInterpolateNonexistentErased() {
        Properties prop = new Properties();
        prop.setProperty("key", "value");
        String text = "This is a test of '${key}' and '${nothing}'";
        String expResults = "This is a test of 'value' and ''";
        String results = InterpolationUtil.interpolate(text, prop);
        assertEquals(expResults, results);
    }
```

Think it would be good to also explicitly test the erasure of non-existent property-references (as documented in the InterpolationUtil comments), to avoid them being accidentally retained as-is or replaced by `null`.

This addition will surface that the current coding is not not consistent with the documentation (as the result is currently `This is a test of 'value' and '${nothing}'`). So either the interpolation needs to be fixed, or the Javadoc comment needs to be updated to reflect the actual behaviour and my suggested addition needs to be updated to `testInterpolateNonexistentRetained` and prediction of the current actual result.

Reply:
Would also be good to add duplicates for the MSBUILD SyntaxStyle interpolation (copies of the methods, but with braces instead of curly-braces and addition of the SyntaxStyle parameter to the interpolate call